cmd/svcat/binding/unbind_cmd.go

```diff
@@ -83,6 +105,46 @@ func (c *unbindCmd) Run() error {
 	var bindings []types.NamespacedName
 	var err error
 
+	if c.abandon {
+		fmt.Fprintln(c.Output, "This action is not reversible and may cause you to be charged for the broker resources that are abandoned.")
+		if !c.skipPrompt {
+			fmt.Fprintln(c.Output, "Are you sure? [y|N]: ")
+			s := bufio.NewScanner(os.Stdin)
+			s.Scan()
+
+			err = s.Err()
+			fmt.Fprintln(c.Output, err)
+
+			if strings.ToLower(s.Text()) != "y" {
+				err = fmt.Errorf("aborted abandon operation")
+				return err
+			}
+		}
+
+		var removedBindings []types.NamespacedName
+		if c.instanceName != "" {
+			si := &v1beta1.ServiceInstance{ObjectMeta: metav1.ObjectMeta{Name: c.instanceName, Namespace: c.Namespace}}
+			removedBindings, err = c.App.RemoveFinalizersByInstance(si)
+		} else {
+			retrievedBindings, err := c.App.RetrieveBindings(c.Namespace)
```

Comment:
This will remove the finalizers on all the bindings in the whole namespace, which I don't think is what we want. I think you could replace this with a call to getBindingsToDelete(), like the part that actually delete the bindings does. Then you would have to either turn that array of names into an array of v1beta1 binding objects, or refactor the pkg methods to take names/namespaces instead of v1beta1.bindings. I think I would do the second, but you could make the case for either.